packages/rulesets/src/spectral/functions/allow-nested-if-parent-exist.ts

```diff
@@ -0,0 +1,50 @@
+// RPC Code: RPC-Get-V1-11
+
+function matchAnyPatterns(patterns: RegExp[], path: string) {
+  return patterns.every((p) => p.test(path))
+}
+
+function verifyNestResourceType(path: string) {
+  const patterns = [/^\/subscriptions\/{subscriptionId}\/resourceGroups\/{resourceGroupName}\/providers\/\w+\.\w+\/\w+\/{\w+}\/\w+.*/gi]
+
+  return matchAnyPatterns(patterns, path)
+}
+
+function verifyResourceType(path: string) {
+  const patterns = [/^\/subscriptions\/{subscriptionId}\/resourceGroups\/{resourceGroupName}\/providers\/\w+\.\w+\/\w+\/{\w+}.*/gi]
+
+  return matchAnyPatterns(patterns, path)
+}
+
+export const allowNestedIfParentExist: any = (fullPath: any, _opts: any, ctx: any) => {
+  const swagger = ctx?.documentInventory?.resolved
+
+  if (fullPath === null || typeof fullPath !== "string" || fullPath.length === 0 || swagger === null) {
+    return []
+  }
+
+  const otherPaths = Object.keys(swagger.paths).filter((p: string) => p !== fullPath)
+
+  if (verifyNestResourceType(fullPath)) {
+    let count = 0
+
+    for (const apiPath of Object.values(otherPaths)) {
+      if (verifyResourceType(apiPath)) {
+        if (fullPath.includes(apiPath)) count++
+      }
```

Comment:
can we not exit the loop if we find 1 apiPath(parent path)?
why count++ and run the loop for all the paths?

Reply:
done, thanks for the catch, added break after finding the path.